support/modules-artifact-gen/single-file-artifact-gen

```diff
@@ -155,8 +155,14 @@ echo "$dest_dir" > $dest_dir_file
 # Create single_file file in plain text
 echo "$filename" > $filename_file
 
+set +e;
+STAT_HAS_f=0;
+stat -f %A "${file}" >/dev/null 2>&1;
+[ $? -eq 0 ] && STAT_HAS_f=1;
+set -e;
+
 # Create permissions file in plain text
-if [[ "$OSTYPE" == "darwin"* ]]; then
+if [ "$STAT_HAS_f" == "1" ]; then
```

Comment:
Mmm. I guess this works, but why to check it as an string? I suggest doing a check for integer:

```
[ $STAT_HAS_f -eq 1 ]
```

Reply:
And thanks for changing `[[` with the basic `test`.

Reply:
thanks. all good suggestions.